contracts/libraries/UniswapRouter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISwapRouter} from "../interfaces/ISwapRouter.sol";
+import "./Path.sol";
+
+library UniswapRouter {
+    using Path for bytes;
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    struct SwapParams {
+        bytes path;
+        address recipient;
+        address tokenIn;
+        address tokenOut;
+        uint256 amountIn;
+        uint256 minAmountOut;
+        address router;
+    }
+
+    event Swap(
+        address indexed tokenIn,
+        address indexed tokenOut,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+
+    function decodePath(bytes memory path)
+        internal
+        pure
+        returns (address tokenIn, address tokenOut)
+    {
+        uint24 fee;
+
+        (tokenIn, tokenOut, fee) = path.decodeFirstPool();
+        bool hasMultiplePools = path.hasMultiplePools();
+
+        if (hasMultiplePools) {
+            path = path.skipToken();
+            while (true) {
+                (, tokenOut, ) = path.decodeFirstPool();
+
+                hasMultiplePools = path.hasMultiplePools();
+
+                if (hasMultiplePools) {
+                    path = path.skipToken();
+                } else {
+                    break;
+                }
+            }
+        }
```

Comment:
Could we not simplify this logic a bit? Here is my attempt:

```solidity
while (path.hasMultiplePools()) {
  (, tokenOut, ) = path.decodeFirstPool();
  path = path.skipToken();
}
```

Reply:
Fixed. Works as per your suggestion with slight modification.

```
(address tokenIn, address tokenOut, ) = swapPath.decodeFirstPool();

while (swapPath.hasMultiplePools()) {
    swapPath = swapPath.skipToken();
    (, tokenOut, ) = swapPath.decodeFirstPool();
}
```